emcc.py

```diff
@@ -935,6 +935,11 @@ def phase_setup(options, state, newargs):
     else:
       settings.SUPPORT_LONGJMP = 'emscripten'
 
+  # SDL2 requires eglGetProcAddress() to work.
+  # NOTE: if SDL2 is updated to not rely on eglGetProcAddress(), this can be removed
+  if settings.USE_SDL == 2 or settings.USE_SDL_MIXER == 2 or settings.USE_SDL_GFX == 2:
```

Comment:
It should be enough to just check `USE_SDL` here since `tools/ports/sdl2_mixer.py` and `tools/ports/sdl2_gfx.py` both set this as part of `process_dependencies`

Reply:
check

Reply:
I'm curious as to why we need all three here? Does this run before `process_dependencies`?